lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
     };
 =cut
 
-sub oscap_security_guide_setup {
+sub get_tests_config {
+    # Get the tests configuration file from repository
+    my $config_file_name = "openqa_config.conf";
+    my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
+
+    download_file_from_https_repo($url, $config_file_name);
+
+    my $config_file_path = script_output("pwd", quiet => 1);
+    $config_file_path =~ s/\r|\n//g;
+    $config_file_path .= "/$config_file_name";
+
+    my $data = script_output("cat $config_file_path", quiet => 1);
+    my $config = Config::Tiny->new;
+    $config = Config::Tiny->read_string("$data");
+    my $err = Config::Tiny::errstr;
+    if ($err eq "") {
+        $use_content_type = $config->{tests_config}->{use_content_type};
+        $remove_rules_missing_fixes = $config->{tests_config}->{remove_rules_missing_fixes};
+        $use_exclusions = $config->{tests_config}->{use_exclusions};
+        record_info("Set test configuration", "Set test configuration from file $config_file_path\n use_content_type = $use_content_type\n  remove_rules_missing_fixes = $remove_rules_missing_fixes\n use_exclusions = $use_exclusions");
+    }
+    else {
+        record_info("Tiny->read error", "Config::Tiny->read( $config_file_path )returened error:\n$err");
+    }
+    return $config_file_path;
+}
+
+sub get_test_expected_results {
+    # Get efpected results from remote file
+    my $eval_match = ();
+    my $found = -1;
+    my $type = "";
+    my $arch = "";
+
+    if ($ansible_remediation == 1) {
+        $type = 'ansible';
+    }
+    else {
+        $type = 'bash';
+    }
+    if (is_s390x) { $arch = "s390x"; }
+    if (is_aarch64 or is_arm) { $arch = "aarch64"; }
+    if (is_ppc64le) { $arch = "ppc"; }
+    if (is_x86_64) { $arch = "x86_64"; }
+    # $sle_version and $profile_ID are global varables
+    my $exp_fail_list_name = $sle_version . "-exp_fail_list";
+    my $expected_results_file_name = "openqa_tests_expected_results.yaml";
+    my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
+
+    download_file_from_https_repo($url, $expected_results_file_name);
+    upload_logs("$expected_results_file_name") if script_run "! [[ -e $expected_results_file_name ]]";
+    my $data = script_output("cat $expected_results_file_name", quiet => 1);
+
+    # Pharse the expected results
+    my $expected_results = YAML::PP::Load($data);
+    record_info("Looking expected results", "Looking expected results for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch");
+
+    $eval_match = $expected_results->{$profile_ID}->{$type}->{$arch}->{$exp_fail_list_name};
+    # If results defined
+    if (defined $eval_match) {
+        $found = 1;
+        if (not defined $$eval_match[0]) {
+            my @eval_match = ();
+            $_[0] = \@eval_match;
```

Comment:
this afir in Perl is not recommended, you are changing the input parameters, I tried to follow from the test the expected result and was really complicated.
For this you can google it but I could find for example [this](https://stackoverflow.blog/2022/09/08/this-is-not-your-grandfathers-perl/)

Reply:
Fixed - Simplified data file and functions get_test_exclusions, get_test_expected_results